toolkit/docs/formats/imageconfig.md

```diff
@@ -156,6 +156,50 @@ A sample KernelOptions specifying a default kernel and a specialized kernel for
+
+##### GPL Licensing
+If using a read-only root in conjunction with a verified boot flow that uses a signed initramfs carefully consider the implications on GPLv3 code. The read-only nature of the filesystem means a user cannot replace GPLv3 components without re-signing a new initramfs.
+
+##### Users
+Since users are controlled by files in `/etc` they are read-only when this is set. It is recommended to either use SSH key based login, or pre-hash the password to avoid storing passwords is plain text in the config files (See [Users](#users)).
+
+##### Separate `/boot` Partition
+Since the root partition's hash tree is stored as part of the initramfs, the initramfs cannot be stored on the same root partition (it would invalidate the measurements). To avoid this a separate `/boot` partition is needed to house the hash tree (via the initramfs).
+
+##### ISO
+The ISO installer supports enabling read-only roots if they are configured through the configuration JSON file (see [full.json's](../../imageconfigs/full.json) `"CBL-Mariner Core Read-Only"` entry). The automatic partition creation mode will create the required `/boot` partition if the read-only root is enabled.
+- `Enable`: Enable dm-verity on the root filesystem
+- `Name`: Custom name for the mounted root (default is `"verity_root_fs"`)
+- `ErrorCorrectionEnable`: Enable automatic error correction of modified blocks (default is `true`)
+- `ErrorCorrectionEncodingRoots`: Increase overhead to increase resiliency of the forward error correction (default is `2` bytes of code per 255 bytes of data)
+- `RootHashSignatureEnable`: Validate the root hash against a key stored in the kernel's system keyring. The signature file should be called `<Name>.p7` and must be stored in the initramfs. This signature WILL NOT BE included automatically in the initramfs. It must be included via an out of band build step.
+- `ValidateOnBoot`: Run a validation of the full disk at boot time, normally blocks are validated only as needed. This can take several minutes if the disk is corrupted.
+- `VerityErrorBehavior`: Indicate additional special system behavior when encountering an unrecoverable verity corruption. One of `"ignore"`, `"restart"`, `"panic"`. Normal behavior is to return an IO error when reading corrupt blocks.
+- `TmpfsOverlays`: Mount these paths as writable overlays backed by a tmpfs in memory.
+- `TmpfsOverlaySize`: Maximum amount of memory the overlays may use. Maybe be one of three forms: `"1234"`, `"1234[k,m,g]"`, `"20%"` (default is `"20%"`) 
+- `TmpfsOverlayDebugEnabled`: Make the tmpfs overlay mounts easily accessible for debugging purposes. They can be found in /mnt/verity_overlay_debug_tmpfs. Include the
+    `verity-read-only-root-debug-tools` package to create the required mount points.
+
+A sample ReadOnlyVerityRoot specifying a basic read-only root using default error correction:
```

Comment:
Might be clearer to segment this sample json portion into a new subsection. At the moment, this looks to be tied to the ISO json but this could be added in any imageconfig.json. Or move to the top of the "ReadOnlyVerityRoot" section.